search_service/proxy/elasticsearch.py

```diff
@@ -113,6 +117,102 @@ def _search_helper(self, page_index: int,
                                        client=client,
                                        model=model)
 
+    def _create_document_helper(self, data: List[Table], index: str) -> str:
+        # fetch indices that use our chosen alias (should only ever return one in a list)
+        indices = self._fetch_old_index(index)
+
+        for i in indices:
+            # build a list of elasticsearch actions for bulk upload
+            actions = self._build_index_actions(data=data, index_key=i)
+
+            # bulk create or update data
+            self._bulk_helper(actions)
+
+        return index
+
+    def _update_document_helper(self, data: List[Table], index: str) -> str:
+        # fetch indices that use our chosen alias (should only ever return one in a list)
+        indices = self._fetch_old_index(index)
+
+        for i in indices:
+            # build a list of elasticsearch actions for bulk update
+            actions = self._build_update_actions(data=data, index_key=i)
+
+            # bulk update existing documents in index
+            self._bulk_helper(actions)
+
+        return index
+
+    def _delete_document_helper(self, data: List[str], index: str) -> str:
+        # fetch indices that use our chosen alias
+        indices = self._fetch_old_index(index)
+
+        # set the document type
+        type = 'user' if index is USER_INDEX else 'table'
```

Comment:
is there a way we could make it more generic here? And does the type info get stored in actual ES document?

Reply:
type is specifically required by elasticsearch's bulk API, so we do need to have some way to know which type we are talking about (especially since we have user and table information now in es). I could just default this to table, but I wanted this to work for either user or table documents. People using this code base could update the USER_INDEX value to match whatever it is in their version of es.

Another alternate option is that I make a helper method for getting the type and call that method here (the method would have some sort of switch statement that returns the correct type based on index